apps/speed.c

```diff
@@ -1272,9 +1225,6 @@ static int run_benchmark(int async_jobs, int (*loop_function)(void *), loopargs_
                         total_op_count += job_op_count;
                     }
                     --num_inprogress;
-#if defined(OPENSSL_SYS_UNIX)
-                    FD_CLR(job_fd, &waitfdset);
```

Comment:
Just to follow up on that, the original bug was caused because select behaves differently to epoll and alters the waitfdset on every call. Effectively it only leaves the fd's that have had events on them in waitfdset when the select returns. The original code was trying to be efficient by only adding or removing fd's when as needed and reusing the same waitfdset on each select call. This worked well with the dasync engine as every asynchronous call results in an immediate notify (and this is still true when using multiple jobs) so you get left with all the fd's in waitfdset every call to select. When we tested with a real asynchronous engine, the  jobs may or may not have set events by the time the select completes and we lost the fd's out of waitfdset for the ones that had not set events. The new code correctly refreshes the waitfdset every time round the loop so we don't lose any fd's now. As we refresh everytime it makes sense to just call FD_ZERO on the waitfdset rather than trying to use FD_CLR on an individual fd.